File: zlh/LeetCode/2023-11-21/530. 二叉搜索树的最小绝对差.ts
function getMinimumDifference(root: TreeNode | null): number {
  const treeNodes: number[] = [];
  let min: number = Number.MAX_VALUE;

  const midTraversal = (root: TreeNode | null): null | undefined => {
    if (!root) return null;

    midTraversal(root.left);
    treeNodes.push(root.val);
    midTraversal(root.right);
  };

  // 中序遍历获取有序数组
  midTraversal(root);

  if (treeNodes.length < 2) return 0;

  // 遍历
  for (let i: number = 1; i < treeNodes.length; i++) {
    min = Math.min(min, treeNodes[i] - treeNodes[i - 1]);
  }

  return min;
}

File: zjj/530. 二叉搜索树的最小绝对差.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def getMinimumDifference(self, root: TreeNode[TreeNode]) -> int:
        li = []
        # BTS通过中序遍历得到升序序列

        def dfs(root):
            if not root:
                return
            dfs(root.left)
            li.append(root.val)
            dfs(root.right)
        dfs(root)
        ans = float('inf')
        for i in range(len(li) - 1):
            if ans > li[i + 1] - li[i]:
                ans = li[i + 1] - li[i]
        return ans
